sql/parser/builtins.go

```diff
@@ -66,9 +67,16 @@ func (b builtin) match(types typeList) bool {
 // sorted please.
 var builtins = map[string][]builtin{
 	// TODO(XisiHuang): support encoding, i.e., length(str, encoding).
-	"length": {stringBuiltin1(func(s string) (Datum, error) {
-		return DInt(len(s)), nil
-	}, DummyInt)},
+	"length": {
+		stringBuiltin1(func(s string) (Datum, error) {
+			return DInt(utf8.RuneCountInString(s)), nil
```

Comment:
If we're going to count characters instead of bytes (and I think we should), we have to be consistent: substr(), left(), and right() need to use the same units as length().